src/app/services/campaign.service.ts

```diff
@@ -482,9 +482,9 @@ export class CampaignService {
     self.methods.anyKnownNPCs = (): boolean => self.methods.listKnownNPCs().length > 0
     self.methods.isActiveNPC = (npc: CampaignNpc): boolean => npc.id === self.locals.active_npc
 
-    self.methods.addNPC = (npc: CampaignNpc = new CampaignNpc): void => {
+    self.methods.addNPC = (npc: CampaignNpc = new CampaignNpc, name: string): void => {
```

Comment:
Rather than passing a second parameter into this, we can check to see if the npc object has a name property or not. Newly initialized NPC objects have name set to `null` by default.

Reply:
Great!